### ISSUE TITLE
When embedding botui in iframe, autoscrolling to the bottom doesn't work.

### ISSUE THREAD

**Issue by kyang6** [botui/botui#63]
When embedding botui in an iframe, autoscrolling down to the bottom of the page after a new message doesn't work. Is there a workaround/solution for this? 

Here is a test link I made showing the issue. 

http://ally.include.ai/scrolltest.html

I am not too familiar with Vue but I saw that the _container is used for autoscroll. Not sure how this mechanism is being affected by being embedding in an iframe though!

**Comment by kyang6** [comment]
Yeah! The workaround I made was using Javascript to just scroll all the way to the bottom of the page on inset of a new component. I don’t have the code anymore but if I recall correctly replace the code in botui.js with this:

root.Vue.directive('botui-scroll', {
 inserted: function (el) {
  window.scrollTo(0,document.body.scrollHeight);
 }
});

Let me know if this doesn't work! I can try finding my older code.

Kevin

**Comment by victorbuckservices** [comment]
I have the same issue. Any update on a workaround ?

**Comment by fconrotte** [comment]
I found the solution 

you have to replace

<button type="button" :class="button.cssClass" class="botui-actions-buttons-button" 

with 

<button type="button" :class="button.cssClass" class="botui-actions-buttons-button" v-botui-scroll

and line 214 with

    root.Vue.directive('botui-scroll', {
      inserted: function (el) {
        el.scrollIntoView(true);
      }
    });

**Comment by Hernan-Torres** [comment]
Is there any option for not auto scrolling to the bottom?
My client requires that only scroll when there is any question buttons for user.

**Comment by Alagaesia93** [comment]
I found out a little bug related to this thread and the el.scrollIntoView(true), but I'm not sure if without it would be ok.

If you have some stuff below the botui component in the page the scroll into view makes the active field appear on top of the page, losing all the other intermediate messages

Example (with bootstrap - try on a very little screen)

```
<div class="row">
  <div class="col">
    <bot-ui />
  </div>
</div>
<div class="row" style="height: 400px">
  <div class="col">
    <h1>Something else</h1>
  </div>
</div>
```

The bot schema could be:
- welcome message
- buttons
- other message
- text field

When you click on the button the page scrolls up to the text field, and the user must scroll up to see the second message.

Thank you for your work, much appreciated!

**Comment by palespectre** [comment]
i ran into the same issue and i found a workaround.

if you want to prevent botui from auto scrolling to the bottom, you just have to remove the scrollIntoView function :


    root.Vue.directive('botui-scroll', {
      inserted: function (el) {
        _container.scrollTop = _container.scrollHeight;
	      el.scrollIntoView(true);
      }
    });

But if you want a nice, smooth and centered scroll, just call scrollIntoView() with these parameters :

    root.Vue.directive('botui-scroll', {
      inserted: function (el) {
        _container.scrollTop = _container.scrollHeight;
	      el.scrollIntoView({behavior: "smooth", block: "center", inline: "nearest"});
      }
    });

**Comment by botviet** [comment]
My fix:

root.Vue.directive('botui-scroll', {
      inserted: function (el) {
        _container.scrollTop = _container.scrollHeight;
        el.scrollTo(0, el.scrollHeight);
      }
    });
